pkg/netpol/ingressanalyzer/ingress_analyzer.go

```diff
@@ -156,29 +203,43 @@ func (ia *IngressAnalyzer) AllowedIngressConnections() map[string]eval.Connectio
 	// then we have ingress connections to that peer
 
 	// get all targeted workload peers and compute allowed conns of each workload peer
+	// 1. from routes
+	routesResult := ia.allowedIngressConnectionsByIngressObjects(scan.Route)
+	// 2. from k8s-ingress objects
+	ingressResult := ia.allowedIngressConnectionsByIngressObjects(scan.Ingress)
+
+	// merge the map results from routes and k8s-ingress objects
+	mergeResults(routesResult, ingressResult)
+	return routesResult
+}
+
+func (ia *IngressAnalyzer) allowedIngressConnectionsByIngressObjects(objType string) map[string]eval.Connection {
+	mapToIterate := ia.routesToServicesMap
+	if objType == scan.Ingress {
+		mapToIterate = ia.k8sIngressToServicesMap
+	}
 	res := make(map[string]eval.Connection)
-	for ns, rtSvcMap := range ia.routesToServicesMap {
+	for ns, objSvcMap := range mapToIterate {
 		// if there are no services in same namespace of the route, the routes in this ns will be skipped
 		if _, ok := ia.servicesToPeersMap[ns]; !ok {
 			continue
 		}
 
-		for _, svcList := range rtSvcMap {
-			routeTargetPeers := ia.getRouteTargetedPeers(ns, svcList)
+		for _, svcList := range objSvcMap {
+			ingressObjTargetPeers := ia.getIngressObjectTargetedPeers(ns, svcList)
 			// avoid duplicates in the result
-			for _, peer := range routeTargetPeers {
+			for _, peer := range ingressObjTargetPeers {
```

Comment:
to get `peerStr` in the next line, can use `peer.String()` instead of having to use `types.NamespacedName`